docassemble/AssemblyLine/data/sources/languages.yml

```diff
@@ -0,0 +1,141 @@
+en: 
+  name: English
+  native_name: English
+es:
+  name: Spanish
+  native_name: Español
+zh: 
+  name: Chinese
+  native_name: 中文
+fr: 
+  name: French
+  native_name: Français
+de: 
+  name: German
+  native_name: Deutsch
+ru: 
+  name: Russian
+  native_name: Русский
+hi: 
+  name: Hindi
+  native_name: हिन्दी
+ar: 
+  name: Arabic
+  native_name: العربية
+ja: 
+  name: Japanese
+  native_name: 日本語
+ko: 
+  name: Korean
+  native_name: 한국어
+vi: 
+  name: Vietnamese
+  native_name: Tiếng Việt
+it: 
+  name: Italian
+  native_name: Italiano
+pt: 
+  name: Portuguese
+  native_name: Português
+tl: 
+  name: Tagalog
+  native_name: Wikang Tagalog
+pl: 
+  name: Polish
+  native_name: Polski
+nl: 
+  name: Dutch
+  native_name: Nederlands
+el: 
+  name: Greek
+  native_name: Ελληνικά
+ht: 
+  name: Haitian Creole
+  native_name: Kreyòl ayisyen
+zh-Hans: 
+  name: Mandarin (Simplified)
+  native_name: 普通话
+zh-Hant: 
+  name: Cantonese (Traditional)
+  native_name: 粵語
```

Comment:
```suggestion
zh-Hans: 
  name: Mandarin (Simplified)
  native_name: 简体中文
zh-Hant: 
  name: Mandarin (Traditional)
  native_name: 繁體中文
zh-HK:
  name: Cantonese (Simplified)
  native_name:  粤语 (简体)
```

ChatGPT def got this wrong; I replaced it with the characters used in Android's language selection.  But there are a few things we should figure out:

* Cantonese has a few different language codes; `yue` in ISO 639-3 seems to be the best one, but [zh-HK](https://sites.psu.edu/symbolcodes/languages/asia/chinese/) does seem to be recommended for some. Neither distinguish between traditional and simplified script, even though that is distinguished in the Android selection that I checked.
* If this list is primarily for choosing a language, I suggest that we remove `Chinese` from the top, and use these more specific ones, as they're a bit confusing, and mostly mean the same thing.  If we do, we might want to use `Chinese` and not `Mandarin`

Reply:
We should feel free to have as many unique variations as we want. The author specifies the list of language codes that they want to use and this is just a lookup table for the native language version of the name. So I don't suggest we remove "Chinese," especially as the written form of Chinese is distinct and can be either traditional or simplified. Most Cantonese speakers use the traditional script and most Mandarin speakers us the simplified, but it's not totally 1 to 1.

In my existing interviews I use zh-t for traditional and I let users do Google Translate for switching between the two. Unlike character based languages, Chinese can be 100% machine translated, at least from info I received from the Asian Outreach Unit at GBLS. So it's less likely that an author will translate one interview into both systems.

The goal with the YAML is to have a good list of potential ISO/custom codes that authors might use.

Reply:
Also found this reference: https://stackoverflow.com/questions/18902072/what-standard-do-language-codes-of-the-form-zh-hans-belong-to

Reply:
> So I don't suggest we remove "Chinese," especially as the written form of Chinese is distinct and can be either traditional or simplified

This statement seems contradictory; because there are two different forms, I don't think we should the use a label that's more vague, rather than one of the two, more specific, labels. Like you said, we want "to have as many unique variations as we want", but `Chinese` by itself isn't really unique, it's an umbrella for a lot of variation.

That does make sense that someone won't translate one interview into both, so that's fair. But there's benefit in the label matching which translation was used.

FWIW, this is from the [RFC mentioned in your last link](https://www.rfc-editor.org/rfc/pdfrfc/rfc5646.txt.pdf)

> The language subtag ’zh’ can still be used without an extended language subtag to label a resource as some unspecified variety of Chinese, while the primary language subtag (’gan’, ’yue’, ’cmn’) is preferred to using the extended language form ("zh-gan", "zh-yue", "zh-cmn")

So, should probably use `zh-cmn-Hant`, `zh-cmn-Hans`, and `zh-yue`, maybe with extensions as well.

Reply:
It doesn't make a lot of sense AFAIK to talk about the written language (which is what Docassemble offers) as either Mandarin or Cantonese. So I think the `cmn` and `yue` would add confusion. That said I don't speak any Chinese language so I could be totally off base.

Reply:
Thinking this over, I believe the description should be Chinese (simplified) and Chinese (traditional) and the tags should be zh-Hans and zh-Hant, without mentioning Cantonese or Mandarin.

Reply:
That's probably fine. I had been basing a lot of my suggestions on the language options present in Android, but after looking at what Windows and Ubuntu have for language selections, I think you're right. I guess Android also has some spoken portions that would necessitate a difference between Cantonese and Mandarin. Sorry!